zipline/pipeline/factors/factor.py

```diff
@@ -433,14 +438,103 @@ def __repr__(self):
 
 
 class CustomFactor(RequiredWindowLengthMixin, CustomTermMixin, Factor):
-    """
-    Base class for user-defined Factors operating on windows of raw data.
+    '''
+    Base class for user-defined Factors.
 
-    TODO: This is basically the most important class to document in the whole
-    Pipeline API...
+    Parameters
+    ----------
+    inputs : iterable, optional
+        An iterable of `BoundColumn` instances (e.g. USEquityPricing.close),
+        describing the data to load and pass to `self.compute`.  If this
+        argument is passed to the CustomFactor constructor, we look for a
+        class-level attribute named `inputs`.
+    window_length : (optional), int
+        Number of rows of rows to pass for each input.  If this
+        argument is passed to the CustomFactor constructor, we look for a
+        class-level attribute named `window_length`.
 
-    We currently only support CustomFactors of type float64.
-    """
+    Notes
+    -----
+    Users implementing their own Factors should subclass CustomFactor and
+    implement a method named `compute` with the following signature::
+
+        def compute(self, today, assets, out, *inputs):
+           ...
+
+    On each simulation date, ``compute`` will be called with the current date,
+    an array of sids, an output array, and an input array for each expression
+    passed as inputs to the CustomFactor constructor.
+
+    The specific types of the values passed to `compute` are as follows::
+
+        today : np.datetime64[ns]
+            Row label for the last row of all arrays passed as `inputs`.
+        assets : np.array[int64, ndim=1]
+            Column labels for `out` and`inputs`.
+        out : np.array[float64, ndim=1]
+            Output array of the same shape as `assets`.  `compute` should write
+            its desired return values into `out`.
+        *inputs : tuple of np.array
+            Raw data arrays corresponding to the values of `self.inputs`.
+
+    ``compute`` functions should expect to be passed NaN values for dates on
+    which no data was available for an asset.  This may include dates on which
+    an asset did not yet exist.
+
+    For example, if a CustomFactor requires 10 rows of close price data, and
+    asset A started trading on Monday June 2nd, 2014, then on Tuesday, June
+    3rd, 2014, the column of input data for asset A will have 9 leading NaNs
+    for the preceding days on which data was not yet available.
+
+    Examples
+    --------
+
+    A CustomFactor with pre-declared defaults::
```

Comment:
This makes the class definition here render as quoted/monospace text in our generated docs.